src/modules/app/routes/categoriesRedirect.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import categoriesFetcher from '../../endpoint/endpoints/categories'
+import { createAction } from 'redux-actions'
+import { redirect } from 'redux-first-router'
+
+import type { Dispatch, GetState } from 'redux-first-router/dist/flow-types'
+import { goToCategories } from './categories'
+import { goToNotFound } from './notFound'
+
+export const CATEGORIES_REDIRECT_ROUTE = 'CATEGORIES_REDIRECT'
+export const goToCategoriesRedirect = (city: string, language: string, categoryId: number) =>
+  createAction(CATEGORIES_REDIRECT_ROUTE)({city, language, categoryId})
+
+/**
+ * Route for changing the language in the categories route
+ * @type {{path: string, thunk: function(Dispatch, GetState)}}
+ */
+export const categoriesRedirectRoute = {
+  path: '/:city/:language/redirect/:categoryId',
+  thunk: async (dispatch: Dispatch, getState: GetState) => {
+    const state = getState()
+    const {city, language, categoryId} = state.location.payload
+
+    const categoriesPayload = await categoriesFetcher.loadData(dispatch, state.categories, {city, language})
+
+    if (!categoriesPayload.data) {
+      // todo error handling
```

Comment:
Does this belong to an other task?

Reply:
yep, I would do this in WEBAPP-203 along with all the other error handling stuff